src/common/com/intellij/plugins/haxe/lang/psi/impl/HaxeReferenceImpl.java

```diff
@@ -565,9 +565,11 @@ public boolean isReferenceTo(PsiElement element) {
       if (chain) return false;
     }
     final PsiElement resolve = element instanceof HaxeComponentName ? resolveToComponentName() : resolve();
+    boolean b = resolve instanceof HaxeNamedComponent;
+    boolean b1 = (resolve instanceof HaxeComponentName || b) &&
+                 resolve.getParent() instanceof HaxeClass;
     if (element instanceof HaxeFile &&
-        resolve instanceof HaxeComponentName &&
-        resolve.getParent() instanceof HaxeClass) {
+        (b1 || resolve instanceof AbstractHaxePsiClass) ) {
```

Comment:
Maybe it checks if reference(usage in this case) points to proper file, then perhaps it makes some sense